packages/ember-data/lib/system/store.js

```diff
@@ -1042,21 +1042,22 @@ Store = Service.extend({
 
     @method query
     @param {String} modelName
-    @param {any} query an opaque query to be used by the adapter
+    @param {Hash} options, query an opaque query of the form {query: { any }} to be used by the adapter
     @return {Promise} promise
   */
-  query: function(modelName, query) {
+  query: function(modelName, options) {
+    Ember.assert("You need to pass a query hash in the options hash of the store's query method", options && options.query);
```

Comment:
This is ok for master. We need to make sure that we change this to a deprecation warning when we cherry-pick this commit to the release branch.